sgv2-service-common/src/main/java/io/stargate/sgv2/common/grpc/DefaultStargateBridgeClient.java

```diff
@@ -0,0 +1,156 @@
+    if (response.hasSchemaChange()) {
+      // Ensure we're not holding a stale copy of the impacted keyspace by the time we return
+      // control to the caller
+      String keyspaceName = addTenantPrefix(response.getSchemaChange().getKeyspace());
+      schema.removeKeyspace(keyspaceName);
+    }
+    return response;
+  }
+
+  @Override
+  public Response executeBatch(Batch request) {
+    return ClientCalls.blockingUnaryCall(
+        channel, StargateBridgeGrpc.getExecuteBatchMethod(), callOptions, request);
+  }
+
+  @Override
+  public CqlKeyspaceDescribe getKeyspace(String keyspaceName) {
+    // TODO authorize?
+    return stripTenantPrefix(schema.getKeyspace(addTenantPrefix(keyspaceName)));
+  }
+
+  @Override
+  public List<CqlKeyspaceDescribe> getAllKeyspaces() {
+    List<String> keyspaceNames = getKeyspaceNames(new ArrayList<>(), null);
+    List<CqlKeyspaceDescribe> keyspaces = new ArrayList<>(keyspaceNames.size());
+    for (String keyspaceName : keyspaceNames) {
+      CqlKeyspaceDescribe keyspace = getKeyspace(keyspaceName);
+      if (keyspace != null) {
+        keyspaces.add(keyspace);
+      }
+    }
+    return keyspaces;
+  }
+
+  private List<String> getKeyspaceNames(List<String> accumulator, BytesValue pagingState) {
+    Query query =
+        (pagingState == null)
+            ? SELECT_KEYSPACE_NAMES
+            : Query.newBuilder(SELECT_KEYSPACE_NAMES)
+                .setParameters(QueryParameters.newBuilder().setPagingState(pagingState).build())
+                .build();
+    ResultSet resultSet = executeQuery(query).getResultSet();
+    for (Row row : resultSet.getRowsList()) {
+      accumulator.add(row.getValues(0).getString());
+    }
+    return (resultSet.hasPagingState())
+        ? getKeyspaceNames(accumulator, resultSet.getPagingState())
+        : accumulator;
+  }
+
+  private Channel addMetadata(Channel channel, String tenantId) {
+    Metadata metadata = new Metadata();
+    metadata.put(HOST_KEY, tenantId);
+    return ClientInterceptors.intercept(
+        channel, MetadataUtils.newAttachHeadersInterceptor(metadata));
+  }
+
+  private String encodeKeyspacePrefix(String tenantId) {
+    return Hex.encodeHexString(tenantId.getBytes(StandardCharsets.UTF_8)) + "_";
+  }
+
+  private String addTenantPrefix(String keyspaceName) {
+    return tenantPrefix + keyspaceName;
+  }
+
+  private CqlKeyspaceDescribe stripTenantPrefix(CqlKeyspaceDescribe describe) {
+    if (describe == null || tenantPrefix.isEmpty()) {
+      return describe;
+    }
+    CqlKeyspace keyspace = describe.getCqlKeyspace();
+    return CqlKeyspaceDescribe.newBuilder(describe)
+        .setCqlKeyspace(
+            CqlKeyspace.newBuilder(keyspace).setName(stripTenantPrefix(keyspace.getName())).build())
+        .build();
+  }
+
+  private String stripTenantPrefix(String fullKeyspaceName) {
+    if (fullKeyspaceName.startsWith(tenantPrefix)) {
+      return tenantPrefix;
```

Comment:
Should this return `tenantPrefix` here? Maybe I'm misunderstanding, but  I thought if `fullKeyspace = "abc_ksname"` and `tenantPrefix = "abc"` this would return `"ksname"`.

Reply:
Good catch, will fix